src/sysext/sysext.c

```diff
@@ -2267,6 +2268,62 @@ static int vl_method_refresh(sd_varlink *link, sd_json_variant *parameters, sd_v
         return sd_varlink_reply(link, NULL);
 }
 
+static int parse_version(const char *filename, char **ret) {
+        const char *start, *ext_suffix;
+        char *version = NULL;
+        /* We are only looking for extension images here, because for directories, we want a no-op*/
+        static const char *valid_extensions[] = {".confext.raw", ".sysext.raw", ".raw"};
+
+        assert(filename);
+        assert(ret);
+
+        /* Find the last occurrences of '_'. This will mark the start of the extension version. */
+        start = strrchr(filename, '_');
+        if (!start) {
+                /* If no underscore found, assign "none" and return success.
+                * This is for cases where there are no versions in the filename like 'image.raw'. */
+                version = strdup("none");
+                if (!version)
+                        return -ENOMEM;
+                *ret = version;
```

Comment:
This should be split out as a goto label and reused below

Reply:
@bluca can you comment here, should I add that in again?

Reply:
fine to add it back if Mike prefers